pq-crypto/kyber_r3/kyber512r3_kem.c

```diff
@@ -24,7 +24,13 @@
 int s2n_kyber_512_r3_crypto_kem_keypair(unsigned char *pk, unsigned char *sk)
 {
     POSIX_ENSURE(s2n_pq_is_enabled(), S2N_ERR_PQ_DISABLED);
-    POSIX_GUARD(indcpa_keypair(pk, sk));
+    #if defined(KYBER512R3_AVX2_BMI2_FLAGS)
+    if (s2n_kyber512r3_is_avx2_bmi2_enabled()) {
+        POSIX_GUARD(indcpa_keypair_avx2(pk, sk));
+    }
+    #else
+        POSIX_GUARD(indcpa_keypair(pk, sk));
+    #endif
```

Comment:
This logic is not correct. If `KYBER512R3_AVX2_BMI2_FLAGS` is enabled and `s2n_kyber512r3_is_avx2_bmi2_enabled` is NOT, then no function will be called.
(same for the other occurrences of this selection logic)